framework/include/utils/TransitionLayerTools.h

```diff
@@ -0,0 +1,210 @@
+                                      const subdomain_id_type transition_layer_id,
+                                      const boundary_id_type input_boundary_1_id,
+                                      const boundary_id_type input_boundary_2_id,
+                                      const boundary_id_type begin_side_boundary_id,
+                                      const boundary_id_type end_side_boundary_id);
+
+/**
+ * Generates weights, weighted indices and corresponding interpolation
+ * @param vec_node_num number of points/nodes on the given curve
+ * @param boundary_points_vec vector of points on the given curve
+ * @param vec_index unweighted index vector
+ * @param wt weights based on point-to-point distances
+ * @param index weighted index vector
+ * @param sigma Gaussian parameter used for Gaussian blurring of local node density
+ * @param linear_vec_x linear interpolation of x coordinates based on weighted index
+ * @param linear_vec_y linear interpolation of y coordinates based on weighted index
+ * @param spline_vec_l spline interpolation of inter-node length based on weighted index
+ */
+void weightedInterpolator(const unsigned int vec_node_num,
+                          const std::vector<Point> boundary_points_vec,
+                          std::vector<Real> & vec_index,
+                          std::vector<Real> & wt,
+                          std::vector<Real> & index,
+                          const Real sigma,
+                          std::unique_ptr<LinearInterpolation> & linear_vec_x,
+                          std::unique_ptr<LinearInterpolation> & linear_vec_y,
+                          std::unique_ptr<SplineInterpolation> & spline_vec_l);
+
+/**
+ * Generates weighted surrogate index vectors on one side for points of a sublayer curve
+ * @param weighted_surrogate_index weighted surrogate index vector
+ * @param unweighted_surrogate_index unweighted surrogate index vector
+ * @param node_number_vec number of points/nodes on the sublayer curve
+ * @param index weighted index vector of the points on the side
+ * @param wt weight vector of the points on the side
+ * @param boundary_node_num number of points/nodes on the side
+ * @param i index of the sublayer
+ */
+void surrogateGenerator(std::vector<Real> & weighted_surrogate_index,
+                        std::vector<Real> & unweighted_surrogate_index,
+                        const std::vector<unsigned int> node_number_vec,
+                        const std::vector<Real> index,
+                        const std::vector<Real> wt,
+                        const unsigned int boundary_node_num,
+                        const unsigned int i);
+
+/**
+ * Decides whether all the Points of a vector of Points are on the XY plane
+ * @param vec_pts vector of points to be examined
+ * @return whether all the Points are on the XY plane
+ */
+bool isXYPlane(const std::vector<Point> vec_pts);
```

Comment:
I'd rather we have a isCoPlanar routine and add it to MooseMeshUtils if that isnt too much work

Reply:
Will do.